src/modules/enrollment/components/EnrollmentQuickActions.tsx

```diff
@@ -30,7 +30,9 @@ const EnrollmentQuickActions = ({
     enabledFeatures.includes(DataCollectionFeatureRole.Service) &&
     enrollment.access.canEditEnrollments;
 
-  const canEditClient = enrollment.client.access.canEditClient;
+  const canEditClient =
+    enrollment.client.access.canEditClient &&
+    enrollment.client.access.canViewClientName;
```

Comment:
```suggestion
    enrollment.client.access.canViewClientName; # Prevent UI-access to edit client if user lacks name access, to avoid situation where masked name ("Client X") is submitted. See follow-up #187358025
```

Reply:
Want to clarify that we're just preventing the user from accessing it in the UI. It's not actually restricted on the mutation.